[react-native] fixed broken links to react-native docs (http error 404)

WookieFPV opened the pull request:
React Native changed their documentation site: https://reactnative.dev/blog/2020/07/23/docs-update

All the current links to the old documentation site (facebook.github.io) will result in a 404 error page.





___

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://reactnative.dev/blog/2020/07/23/docs-update
- [only doc changes] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [only doc changes] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [only doc changes If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

kuasha420 commented:
@WookieFPV That's a lot of links! I've manually tested a few of them and all seems to be working. Was the replacement done manually or with search-replace?

WookieFPV commented:
@kuasha420 I replaced them with a regex (Webstorm/IDEA regex syntax).
`https\:\/\/facebook\.github\.io\/react\-native\/docs\/(.*)\.html(.*)
replaced with:
https://reactnative.dev/docs/$1$2`

the important part was removing ".html" from the URL because it broke the links.
I tested at least 70% of them if not even all (not sure)

alloy commented:
Thanks! 🙏
